corehq/apps/userreports/tasks.py

```diff
@@ -178,25 +186,52 @@ def _run_report(backend_to_use):
     return objects
 
 
+@periodic_task(
+    run_every=ASYNC_INDICATOR_QUEUE_TIME,
+    queue=settings.CELERY_PERIODIC_QUEUE,
+)
+def queue_async_indicators():
+    start = datetime.utcnow()
+    cutoff = start + ASYNC_INDICATOR_QUEUE_TIME
+    time_for_crit_section = ASYNC_INDICATOR_QUEUE_TIME.seconds - 10
+
+    with CriticalSection(['queue-async-indicators'], timeout=time_for_crit_section):
+        redis_client = get_redis_client().client.get_client()
+        indicators = AsyncIndicator.objects.all()[:10000]
```

Comment:
A bit concerned that this could still end up with thousands of tasks in the celery queue if they aren't getting processed fast enough. There's nothing here to stop us from adding more tasks into the queue.